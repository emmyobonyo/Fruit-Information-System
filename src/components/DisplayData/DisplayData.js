import React from 'react';

function DisplayData() {
  return (
    <div>
      
    </div>
  )
}

export default DisplayData;
