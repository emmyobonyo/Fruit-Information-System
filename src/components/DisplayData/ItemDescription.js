/* eslint-disable import/no-extraneous-dependencies */
import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { fetchFruitData } from '../../Redux/displayData/displayDataSlice';
import Navigation from '../Navigation/Navigation';
import './itemDescription.css';

function ItemDescription({ fruits }) {
  const fruit = fruits;
  const fruitItem = useSelector((state) => state.fruits);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchFruitData(fruit));
  });
  return (
    <div>
      <Navigation title={fruit} />
      { fruitItem.map((fruit) => (
        <div key={fruit.id}>
          <div className="item-header">
            <img src={`./assets/${fruit.name}.png`} alt={fruit.name} />
            <h1>{ fruit.name }</h1>
          </div>
          <h5 className="item-paragraph">{` Some more information about ${fruit.name}'s`}</h5>
          <div className="item-table">
            <table>
              <tr>
                <th>id</th>
                <td>{ fruit.id }</td>
              </tr>
              <tr>
                <th>Genus</th>
                <td>{ fruit.genus }</td>
              </tr>
              <tr>
                <th>Name</th>
                <td>{ fruit.name }</td>
              </tr>
              <tr>
                <th>Family</th>
                <td>{ fruit.family }</td>
              </tr>
              <tr>
                <th>Order</th>
                <td>{ fruit.order }</td>
              </tr>
              <tr>
                <th>Carb %</th>
                <td>{ fruit.carbs }</td>
              </tr>
              <tr>
                <th>Protien %</th>
                <td>{ fruit.protein }</td>
              </tr>
              <tr>
                <th>Fat %</th>
                <td>{ fruit.fat }</td>
              </tr>
              <tr>
                <th>Sugar %</th>
                <td>{ fruit.sugar }</td>
              </tr>
              <tr>
                <th>Calories %</th>
                <td>{ fruit.calories }</td>
              </tr>
            </table>
          </div>
        </div>
      )) }
    </div>
  );
}

ItemDescription.propTypes = {
  fruits: PropTypes.string.isRequired,
};

export default ItemDescription;
