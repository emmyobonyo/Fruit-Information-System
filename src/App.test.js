import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter as Router, MemoryRouter, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import DisplayData from './components/DisplayData/DisplayData';
import ItemDescription from './components/DisplayData/ItemDescription';
import store from './Redux/configureStore';

it('should return the home page', () => {
  const homePage = render(<Router><Provider store={store}><DisplayData /></Provider></Router>);
  expect(homePage).toMatchSnapshot();
  expect(homePage.queryByText(/All Fruits/)).toBeInTheDocument();
});

it('should return the Apple', () => {
  const home = render(<MemoryRouter><Route path="/Apple"><Provider store={store}><ItemDescription fruits="Apple" /></Provider></Route></MemoryRouter>);
  expect(home).toMatchSnapshot();
});

it('should return the Apple', () => {
  const home = render(<MemoryRouter><Route path="/Apple"><Provider store={store}><ItemDescription fruits="Apple" /></Provider></Route></MemoryRouter>);
  expect(home.findByText(/Apple/)).toBeTruthy();
});
