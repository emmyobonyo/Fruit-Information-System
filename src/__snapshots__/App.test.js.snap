// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return the Apple 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should return the home page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <nav
            class="navigation"
          >
            <div
              class="left-navigation"
            >
              <a
                href="/"
              >
                <i
                  class="fas fa-angle-left fa-2x"
                />
              </a>
              <h3 />
            </div>
            <div
              class="search-bar"
            >
              <input
                placeholder="Filter by country"
                type="text"
              />
              <i
                class="fas fa-angle-right fa-2x"
              />
            </div>
            <div
              class="right-navigation"
            >
              <i
                class="fas fa-microphone"
              />
              <i
                class="fas fa-cog"
              />
            </div>
          </nav>
        </div>
        <div
          class="all-fruits"
        >
          <img
            alt="banana"
            src="./assets/banana.png"
          />
          <h1>
            All Fruits
          </h1>
        </div>
        <ul />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <nav
          class="navigation"
        >
          <div
            class="left-navigation"
          >
            <a
              href="/"
            >
              <i
                class="fas fa-angle-left fa-2x"
              />
            </a>
            <h3 />
          </div>
          <div
            class="search-bar"
          >
            <input
              placeholder="Filter by country"
              type="text"
            />
            <i
              class="fas fa-angle-right fa-2x"
            />
          </div>
          <div
            class="right-navigation"
          >
            <i
              class="fas fa-microphone"
            />
            <i
              class="fas fa-cog"
            />
          </div>
        </nav>
      </div>
      <div
        class="all-fruits"
      >
        <img
          alt="banana"
          src="./assets/banana.png"
        />
        <h1>
          All Fruits
        </h1>
      </div>
      <ul />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
